utils.go

```diff
@@ -108,15 +108,35 @@ func formatValue(value driver.Value) (string, error) {
 	case int64, uint64, int32, uint32, int16, uint16, int8, uint8, int, uint:
 		return fmt.Sprintf("%d", value), nil
 	case float64, float32:
-		return fmt.Sprintf("%f", value), nil
+		return fmt.Sprintf("%g", value), nil
 	case bool:
 		if value.(bool) {
 			return "1", nil
 		} else {
 			return "0", nil
 		}
 	case time.Time:
-		return fmt.Sprintf("'%s'", value.(time.Time).Format("2006-01-02 15:04:05 -07:00")), nil
+		// Convert timestamp to UTC and don't add timezone to format
+		// This way we ensure that all Firebolt time types support this string,
+		// while providing the same time data to the engine
+		timeValue := value.(time.Time)
+		layout := "2006-01-02 15:04:05.000000"
+		// Subtract date part from value and check if remaining time part is zero
+		// If it is, use date only format
+		if timeValue.Sub(timeValue.Truncate(time.Hour*24)) == 0 {
+			layout = "2006-01-02"
+		} else if _, offset := timeValue.Zone(); offset != 0 {
+			// If we have a timezone info, add it to format
+			layout = "2006-01-02 15:04:05.000000-07:00"
+		}
+		return fmt.Sprintf("'%s'", timeValue.Format(layout)), nil
+	case []byte:
+		byteValue := value.([]byte)
+		parts := make([]string, len(byteValue))
+		for i, b := range byteValue {
+			parts[i] = fmt.Sprintf("\\x%02x", b)
+		}
+		return fmt.Sprintf("'%s'", strings.Join(parts, "")), nil
```

Comment:
Seems straightforward enough, but I'm surprised there's no function to do so in Golang.

Reply:
Same as in python. You can convert to hex, but in our case we need to put \\x symbol before each byte, seems like that's not standard